### PR TITLE
Added precision for assets. Closes #1169.

### LINKED ISSUE #1169: Add Precision to Asset Quantities
Users should be able to set a precision in the AssetDefinition and then be able to use "fractional" amounts in sending/minting/burning.

### PR DESCRIPTION
**Pull request by Revertron**
### Description of the Change
Added a precision asset type. 

### Benefits

Will give possibility to operate on fractions of a value with set precision.

### CODE REVIEW

File: iroha/src/smartcontracts/isi/asset.rs [hunk 1]
```diff
@@ -159,6 +184,30 @@ pub mod isi {
...
+            assert_asset_type(
+                &self.destination_id.definition_id,
+                wsv,
+                AssetValueType::Fixed,
+            )?;
+            wsv.modify_asset(&self.destination_id, |asset| {
+                let quantity: &mut Fixed = asset.try_as_mut()?;
+                *quantity = quantity
+                    .checked_sub(self.object)
+                    .ok_or_else(|| error!("Not enough quantity to burn."))?;
```
Comment (i1i1) [inline iroha/src/smartcontracts/isi/asset.rs:204]: Could you add this error to math error?
https://github.com/hyperledger/iroha/blob/00ba1876f1f102af141c7f92bcd33273ad601c09/iroha/src/smartcontracts/isi/mod.rs#L105-L109

File: iroha/src/smartcontracts/isi/asset.rs [hunk 2]
```diff
@@ -159,6 +184,30 @@ pub mod isi {
...
+                Ok(())
+            })?;
+            Ok(())
```
Comment (i1i1) [inline iroha/src/smartcontracts/isi/asset.rs:207]:
```suggestion
            })
```

File: iroha_data_model/src/lib.rs [hunk 3]
```diff
@@ -1351,11 +1488,16 @@ pub mod asset {
...
         }
 
-        /// Asset definition with store asset value type.
+        /// Asset definition with big quantity asset value type.
         pub const fn new_big_quantity(id: DefinitionId) -> Self {
             AssetDefinition::new(id, AssetValueType::BigQuantity)
         }
 
+        /// Asset definition with decimal quantity asset value type.
+        pub const fn new_decimal_quantity(id: DefinitionId) -> Self {
```
Comment (i1i1) [inline iroha_data_model/src/lib.rs:1497]:
```suggestion
        pub const fn with_precision(id: DefinitionId) -> Self {
```

File: iroha/src/smartcontracts/isi/asset.rs (hunk 1 above)
Comment (Revertron) [reply iroha/src/smartcontracts/isi/asset.rs]: Done.

File: iroha_data_model/src/lib.rs (hunk 3 above)
Comment (Revertron) [reply iroha_data_model/src/lib.rs]: Done.

File: iroha/src/smartcontracts/isi/asset.rs (hunk 2 above)
Comment (Revertron) [reply iroha/src/smartcontracts/isi/asset.rs]: Changed. And all similar places in this module.

**Review by i1i1 — APPROVED** [review]

**Review by eadventurous — APPROVED** [review]

**Review by i1i1 — APPROVED** [review]

**Pull request merged**